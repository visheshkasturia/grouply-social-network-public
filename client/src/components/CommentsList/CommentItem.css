.commentItem {
  display: grid;
  grid-template-column: 33% 33% 34%;
  grid-template-rows: 65% 35%;
  height: 200px;
  margin-top: 20px;
  margin-bottom: 20px;
  border: solid 2px black;
  border-radius: 10px;
  background: #cbf5d3;
}

.comment-content {
  grid-row: 1/ span 1;
  grid-column: 1/ span 3;
  margin: 10px;
}

.comment-author {
  grid-row: 2/ span 1;
  grid-column: 1/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
}

.comment-edit {
  grid-row: 2/ span 1;
  grid-column: 2/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
}

.comment-delete {
  grid-row: 2/ span 1;
  grid-column: 3/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
}