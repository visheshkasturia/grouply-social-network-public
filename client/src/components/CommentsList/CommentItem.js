/* eslint-disable */
import React, { useState } from 'react';
import './CommentItem.css';
import deleteCommentByCid from '../../apiCalls/deleteCommentByCid';
import PopUp from '../PopUp/PopUp';
import EditComment from '../EditComment';

function CommentItem({comment, handleCommentChanged, updateCommentInPost}) {
  const [edit, setEdit] = useState(false);

  const handleDelete = async () => {
    await deleteCommentByCid(comment._id);
    handleCommentChanged();
    updateCommentInPost();
  }


  const handleEdit = () => {
    setEdit(!edit);
    updateCommentInPost();
  }
  
  return (
    <div className="commentItem">
      <div className="comment-content"> {comment.commentContent}</div>
      <div className="comment-author"> Author: {comment.commentBy }</div>
      <div className="comment-edit"> 
        <button onClick={handleEdit}> Edit Comment </button>
        {edit && <PopUp content = {<EditComment commentId={comment._id}/>} handleClose={handleEdit}/>}
      </div>
      <div className="comment-delete">
        <button onClick={handleDelete}> Delete Comment </button>
      </div>
    </div>
  );
}

export default CommentItem;