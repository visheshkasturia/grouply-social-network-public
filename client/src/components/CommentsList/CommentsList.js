/* eslint-disable */
import React, { useEffect, useState } from 'react';
import CommentItem from './CommentItem';


function CommentsList({comments, updateCommentInPost}) {
  const [commentChanged, setCommentChanged] = useState(false);

  function handleCommentChanged() {
    setCommentChanged(!commentChanged);
  }

  useEffect(() => {
    // setCommentChanged(!commentChanged);
  }, [commentChanged])

  

  return (
    <div>
      { comments.map((comment, i) => (
        <CommentItem key={i} comment={comment} commentChanged={commentChanged} handleCommentChanged={handleCommentChanged} updateCommentInPost={updateCommentInPost}/>
      ))}
    </div>
  );
}

export default CommentsList;