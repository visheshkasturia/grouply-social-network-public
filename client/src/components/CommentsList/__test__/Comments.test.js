import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CommentItem from '../CommentItem'
import CommentsList from '../CommentsList'

const comment = {
  commentBy: '61abcfdf692a1b833a551528',
  commentContent: 'Test- add comment BE - 1??'
}

const comments = [comment]
it('CommentItem renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CommentItem comment={comment} />, div);
});

it('CommentItem matches snapshot', () => {
  const tree = renderer.create(<CommentItem comment={comment}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('CommentsList renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CommentsList comments={comments} />, div);
});


it('CommentsList matches snapshot', () => {
  const tree = renderer.create(<CommentsList comments={comments}/>).toJSON();
  expect(tree).toMatchSnapshot();
});