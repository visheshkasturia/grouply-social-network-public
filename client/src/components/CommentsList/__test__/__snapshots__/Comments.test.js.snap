// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentItem matches snapshot 1`] = `
<div
  className="commentItem"
>
  <div
    className="comment-content"
  >
     
    Test- add comment BE - 1??
  </div>
  <div
    className="comment-author"
  >
     Author: 
    61abcfdf692a1b833a551528
  </div>
  <div
    className="comment-edit"
  >
    <button
      onClick={[Function]}
    >
       Edit Comment 
    </button>
  </div>
  <div
    className="comment-delete"
  >
    <button
      onClick={[Function]}
    >
       Delete Comment 
    </button>
  </div>
</div>
`;

exports[`CommentsList matches snapshot 1`] = `
<div>
  <div
    className="commentItem"
  >
    <div
      className="comment-content"
    >
       
      Test- add comment BE - 1??
    </div>
    <div
      className="comment-author"
    >
       Author: 
      61abcfdf692a1b833a551528
    </div>
    <div
      className="comment-edit"
    >
      <button
        onClick={[Function]}
      >
         Edit Comment 
      </button>
    </div>
    <div
      className="comment-delete"
    >
      <button
        onClick={[Function]}
      >
         Delete Comment 
      </button>
    </div>
  </div>
</div>
`;
